index.js

```diff
@@ -41,5 +41,24 @@ function md5File (filename, cb) {
   input.pipe(output)
 }
 
+function md5BufferSync (buffer) {
+  const hash = crypto.createHash('md5')
+  const total = buffer.length
+
+  let begin = 0
+
+  do {
+    // slice(begin, end)
+    // if end is greater than the length, only to the real end
+    const read = buffer.slice(begin, BUFFER_SIZE)
```

Comment:
Why are you slicing the buffer? I don't think that that would have a performance increase at all, in fact, it probably has the opposite 🤔 

Have you measured this?

Reply:
Sorry, I just tested it and got the right result.

I just imitate the sync method.

Reply:
The sync method was made this way to avoid reading the entire file from disk before using the data, thus avoiding excessive memory consumption. Since the buffer already is a chunk of memory, there is no need to split it up anymore ☺️

Reply:
thanks for your explain😀